File: UI/src/app/models/event-model.ts
export class EventModel {
  /**
   *
   */
  constructor(
    id: string,
    title: string,
    description: string,
    price: number,
    location: string,
    date: string
  ) {
    this.Id = id;
    this.Title = title;
    this.Description = description;
    this.Price = price;
    this.Location = location;
    this.Date = date;
  }

  Id!: string;
  Title!: string;
  Description!: string;
  Price!: number;
  Location!: string;
  Date!: string;
}

File: API/Models/EventModel.cs
namespace API.Models
{
    public record EventModel
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string Location { get; set; }
        public DateTimeOffset Date { get; set; }
    }
}
